magento1-module/app/code/local/Divante/VueStorefrontBridge/controllers/UserController.php

```diff
@@ -321,29 +323,61 @@ public function meAction(){
                         foreach($updatedCustomer['addresses'] as $updatedAdress) {
                             $updatedAdress['region'] = $updatedAdress['region']['region'];
 
-                            if($updatedAdress['default_billing']) {
+                            if ($updatedAdress['delete'] === true) {
+                                try {
+                                    $sAddress = Mage::getModel('customer/address');
+
+                                    $sAddress->load($updatedAdress['entity_id']);
+
+                                    $sAddress->setData($updatedAdress)->delete();
```

Comment:
I'm curious to know why the data is set here before deleting?

Reply:
This is fixed now. 

Copy and paste can lead to bad code :/.